### PR TITLE
[codex] Add Canton response helpers

### PR DESCRIPTION
**Pull request by HardlyDifficult**
## Summary
- Add `normalizeCantonContractItem` and `findCantonCoinBalance` to the shared `canton-response-utils` utility surface.
- Export the response utility module from `src/utils` so callers can import the helpers from `@fairmint/canton-node-sdk`.
- Add focused unit coverage for flat/wrapped active-contract normalization and Amulet balance selection.

## Rationale
These helpers normalize Canton ledger and Amulet response shapes and are not Privy-specific. Moving them to `@fairmint/canton-node-sdk` gives Privy SDK consumers a better shared home without re-expanding the Privy SDK API.

## Validation
- `npm test -- --runInBand test/unit/canton-response-utils.test.ts`
- `npx eslint src/utils/canton-response-utils.ts src/utils/index.ts test/unit/canton-response-utils.test.ts`
- `npx prettier --check src/utils/canton-response-utils.ts src/utils/index.ts test/unit/canton-response-utils.test.ts`
- `npx tsc --noEmit --target ES2020 --module commonjs --lib ES2020,DOM --strict --esModuleInterop --skipLibCheck --exactOptionalPropertyTypes --noUncheckedIndexedAccess --noPropertyAccessFromIndexSignature --noUnusedLocals --noUnusedParameters src/utils/canton-response-utils.ts`

## Local notes
- `npm run build:lint` is blocked in this fresh worktree by missing generated OpenAPI/client files and uninitialized submodules.
- `npm run format` reports pre-existing formatting issues outside the touched files.

---

> [!NOTE]
> **Low Risk**
> New read-only parsing/selection utilities with defensive defaults; no changes to auth, signing, or transaction submission paths.
> 
> **Overview**
> Adds shared **Canton response helpers** on `canton-response-utils` and re-exports them from `src/utils` so `@fairmint/canton-node-sdk` consumers can normalize ledger payloads and pick Canton Coin balances without Privy-specific APIs.
> 
> **`normalizeCantonContractItem`** maps flat or `contractEntry.JsActiveContract.createdEvent`-wrapped active-contract items into a stable `CantonNormalizedContract` shape, with defensive empty defaults on bad input. **`findCantonCoinBalance`** selects the Amulet token for a given DSO/admin (`expectedAdmin`), preferring exact `Amulet` over a case-insensitive `amulet` fallback.
> 
> Typed wallet/UTXO interfaces accompany the helpers. Coverage includes unit tests for normalization edge cases and Amulet selection rules, plus LocalNet integration tests against live ledger and validator wallet data.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b617a9f129b17d93bc183b97841d236c018e613e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **Refactor**
  * Added utilities for improved handling of Canton contract creation data and wallet balance information.

* **Tests**
  * Added comprehensive unit test coverage for Canton data processing utilities.

### CODE REVIEW

File: test/unit/canton-response-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
+import {
+  findCantonCoinBalance,
+  normalizeCantonContractItem,
+  type CantonWalletBalances,
+} from '../../src/utils/canton-response-utils';
+
+describe('canton-response-utils', () => {
+  describe('normalizeCantonContractItem', () => {
```
[the remark this answers is not in the document]
Comment (HardlyDifficult) [reply test/unit/canton-response-utils.test.ts]: Fixed in `8c2bd3a` by adding explicit `: void` return annotations to all `describe`/`it` callbacks in `test/unit/canton-response-utils.test.ts`.

Validation passed:
- `npm test -- --runInBand test/unit/canton-response-utils.test.ts`
- `npx eslint test/unit/canton-response-utils.test.ts src/utils/canton-response-utils.ts src/utils/index.ts`
- `npx prettier --check test/unit/canton-response-utils.test.ts src/utils/canton-response-utils.ts src/utils/index.ts`
- targeted strict `npx tsc --noEmit ... src/utils/canton-response-utils.ts`
